recipes-devtools/rust/rust.inc

```diff
@@ -192,8 +217,27 @@ def llvm_cpu(d):
     except:
         return trans.get(target, "generic")
 
-TARGET_LLVM_CPU="${@llvm_cpu(d)}"
-TARGET_LLVM_FEATURES = "${@llvm_features_from_tune(d)} ${@llvm_features_from_cc_arch(d)}"
+# generates our target CPU value for cross builds
+def llvm_cross_cpu(d):
+    feat = d.getVar('TUNE_FEATURES', True)
+    if not feat:
+        return ""
+    feat = frozenset(feat.split())
+
+    # The only reliable way of finding the cpu is to look in TUNE_FEATURES
+    # which is derived from DEFAULTTUNE
+    if "mips32r2" in feat:
+        return "mips32r2"
+    elif "mips32" in feat:
+        return "mips32"
+
+    # TODO: Proper x86 targets. "generic" is a valid choice.
+    # TODO: Proper ARM cpu selection, for now "generic" is a valid cpu choice.
+    return "generic"
+
+TARGET_LLVM_CPU="${@llvm_cross_cpu(d)}"
```

Comment:
I'm not quite sure I understand. PACKAGE_ARCH is always set to x86_64 (note the underscore) for me, no matter if I'm building rust-cross-mipsel or rust-native. I assume this is because rust-cross-mipsel is still a x86_64 package. I'm not sure that PACKAGE_ARCH will ever result in anything usable by the trans["..."] dict.

TRANSLATED_TARGET_ARCH is pretty boring as it will result in something like "mipsel" for my target which isn't really enough to do anything.